cmd/nerdctl/run_network.go

```diff
@@ -164,12 +168,72 @@ func generateNetOpts(cmd *cobra.Command, dataStore, stateDir, ns, id string) ([]
 				ports = append(ports, pm...)
 			}
 		}
+	case nettype.Container:
+		if runtime.GOOS != "linux" {
+			return nil, nil, "", nil, fmt.Errorf("currently '--network=container:<container>' can only works on linux")
+		}
+		if rootlessutil.IsRootless() {
+			return nil, nil, "", nil, fmt.Errorf("currently '--network=container:<container>' can't run rootless")
+		}
+		if len(netSlice) > 1 {
+			return nil, nil, "", nil, fmt.Errorf("only one network allowed using '--network=container:<container>'")
+		}
+		network := strings.Split(netSlice[0], ":")
+		if len(network) != 2 {
+			return nil, nil, "", nil, fmt.Errorf("invalid network: %s, should be \"container:<id|name>\"", netSlice[0])
+		}
+		containerName := network[1]
+		client, ctx, cancel, err := newClient(cmd)
+		if err != nil {
+			return nil, nil, "", nil, err
+		}
+		defer cancel()
+
+		path, err := getNetworkNamespace(ctx, client, containerName)
+		if err != nil {
+			return nil, nil, "", nil, err
+		}
```

Comment:
We should save the container's ID in the label instead of the original args.

`container:test` -> `container:<id>`